rust/macros/lib.rs

```diff
@@ -2,8 +2,15 @@
 
 //! Crate for all kernel procedural macros.
 
+/// helper functions
+mod helpers;
+
+/// generic module/driver macros
```

Comment:
Please capitalize like the rest of the code, end with period, etc.

Also, all these refactors should probably go first in another PR.

Reply:
This refectory is there, because of need created here. It's needed now, as the rtnl macro is not a module macro. What about not a pr, but a commit?

Reply:
Finn, I think you need to break this whole thing up into separate a bunch of PRs. This is so that we can carefully review this in increments rather than in one huge go.

For example, and as Miguel has already suggested, changes to existing code should be factored out so that they can be reviewed separately. In fact, pure refactors without functional changes should also have their own PRs. Even if they are not immediately needed, you can say something to the effect of "This is in preparation for XXX because XXX" -- see for example #386, where there are no users yet of an enhancement, they come in a subsequent PR. See also #239 as an example of a pure refactor. They (and many others) were broken into different PRs to make reviewing easier.

Reply:
Ok, I will try to add all open reviews on this I to this one, then keep this one open, but without the intention to merge as a meta, and create small prs

Reply:
Thanks Finn -- I know it takes time on the submitter's side, but it is best for everyone and reduces the overall time to get things merged.

Reply:
No problem. But I think I can't do it until this weekend